indexer/indexer.py

```diff
@@ -0,0 +1,337 @@
+                                      ignore=[400])
+        # Loads mappings for the index
+        self.es_client.put_mapping(index=self.index_name,
+                                   doc_type=self.doc_type,
+                                   body=self.create_mapping())
+
+
+class FileIndexer(Indexer):
+    """Create a file oriented index.
+
+    FileIndexer makes use of its index() function to perform indexing
+    of the data files that are presented to the class when creating an
+    instance.
+
+    End for now.
+
+    """
+
+    def index(self, bundle_uuid, bundle_version, **kwargs):
+        """Indexes the data files.
+
+        Triggers the actual indexing process
+
+        :param bundle_uuid: The bundle_uuid of the bundle that will be indexed.
+        :param bundle_version: The bundle of the version.
+        """
+        # Get the config driving indexing (e.g. the required entries)
+        req_entries = self.index_mapping_config['requested_entries']
+        # Iterate over each file
+        for _file in self.data_files.values():
+            # List the arguments for clarity
+            args = [req_entries, "", self.metadata_files]
+            # Get all the contents from the entries requested in the config
+            contents = {key: value for key, value in self.__get_item(*args)}
+            # Get the elasticsearch uuid for this particular data file
+            es_uuid = "{}:{}".format(bundle_uuid, _file['uuid'])
+            # Get the special fields added to the contents
+            special_ = self.special_fields(_file,
+                                           contents,
+                                           bundle_uuid=bundle_uuid,
+                                           bundle_version=bundle_version,
+                                           es_uuid=es_uuid)
+            contents = {**contents, **special_}
+            # Load the current file in question
+            # Ideally merge() should be called at this point
+            self.load_doc(doc_contents=contents, doc_uuid=es_uuid)
+
+    def merge(self, doc_contents):
+        """
+        Merge the document with the contents in ElasticSearch.
+
+        merge() should take the results of get_item() and harmonize it
+        with whatever is present in ElasticSearch to avoid blind overwritting
+        of documents. Users should implement their own protocol.
+
+        :param doc_contents: Current document to be indexed.
+        :return: The harmonized document which can overwrite existing entry.
+        """
+        # Assuming a file is never really changed, there is no reason
+        # for merge here.
+        pass
+
+    def create_mapping(self, **kwargs):
+        """
+        Return the mapping as a string.
+
+        Pulls the mapping from the index_mapping_config.
+        """
+        # Return the es_mapping from the index_mapping_config
+        mapping_config = self.index_mapping_config['es_mapping']
```

Comment:
Consider changing this method's signature to be transformative such that implementors don't have to duplicate the code that extracts the mapping from the config. I can explain in person why that would be preferable.
```
def transform_mapping(self, mapping:dict) -> dict:
    …
```